cli/azd/pkg/project/service_manager.go

```diff
@@ -311,6 +264,19 @@ func (sm *serviceManager) Package(
 	buildOutput *ServiceBuildResult,
 ) *async.TaskWithProgress[*ServicePackageResult, ServiceProgress] {
 	return async.RunTaskWithProgress(func(task *async.TaskContextWithProgress[*ServicePackageResult, ServiceProgress]) {
+		cachedResult, ok := sm.getOperationResult(ctx, serviceConfig, string(ServiceEventPackage))
+		if ok && cachedResult != nil {
+			task.SetResult(cachedResult.(*ServicePackageResult))
+			return
+		}
+
+		if buildOutput == nil {
```

Comment:
Should we skip this if there isn't a build requirement?

Reply:
If build is not required, then this result will be nil.